backend/src/models/likes.js

```diff
@@ -34,13 +34,46 @@ class Likes {
      * Put likes
      */
     putLikes({ type, id, userID, value }) {
-        return this.db
-            .insertUnique('likes', {
-                objectType: type,
-                objectID: id,
-                userID,
-                value
-            })
+
+        // Stuff to update likes
+        const data = {
+            objectType: type,
+            objectID: id,
+            userID,
+            value
+        }
+        const insertValues = Object.values(data)
+        const insertColumns = Object.keys(data)
+        const insertPlaceholders = insertColumns.map(_ => '?').join()
+        const updateLikes = `REPLACE INTO likes (${insertColumns}) VALUES (${insertPlaceholders})`
+
+        // Stuff to update reputation
+        const creatorTable = (type === 'question' || type === 'review') ? type : 'comment'
+        const updateReputation = `UPDATE user
+        SET reputation = (SELECT reputation FROM user WHERE id = ?) + ?
+        WHERE id = ?`
+
+        // Classic
+        const db = this.db._db
+        return new Promise((resolve, reject) => {
+            Promise.all([
+                this.db.query('SELECT * FROM likes WHERE objectType=? AND objectID=? AND userID=?',
+                    [type, id, userID]),
+                this.db.query(`SELECT userID FROM ${creatorTable} WHERE id = ?`,
+                    [id])
+            ])
+                .then(([originalLike, creator]) => {
+                    const oldLike = originalLike && originalLike.value || 0
+                    const creatorID = creator.userID
+                    const repChange = creatorID != userID ? (value - oldLike) : 0
+                    db.serialize(function() {
```

Comment:
I could be wrong, but I'm pretty sure `serialize` does not protect from races between different queries using the same connection - i.e. other node async methods.
It just ensures that one finishes before the next one starts...

Reply:
I'm pretty sure updating the user table and updating the likes table don't have to be race-free.. but yeah they both have to succeed or fail I guess.

Reply:
I think don't worry about races rn... it's not worth the effort and won't matter for the demo.

Reply:
Yeah so its the fact that we guarantee 'BEGIN TRANSACTION' to happen first, but there is still a race cause this isn't all one query.